backend/app/notify/email.go

```diff
@@ -0,0 +1,156 @@
+package notify
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	log "github.com/go-pkgz/lgr"
+	"github.com/pkg/errors"
+	"gopkg.in/gomail.v2"
+)
+
+// EmailParams contain settings for email set up
+type EmailParams struct {
+	Server    string
+	Port      int
+	From      string
+	Username  string
+	Password  string
+	KeepAlive time.Duration
+}
+
+// Email implements notify.Destination for email
+type Email struct {
+	server    string
+	port      int
+	from      string
+	username  string
+	password  string
+	keepAlive time.Duration
+	sendChan  chan *gomail.Message
+	errChan   chan error
+}
+
+const emailConnectionKeepAlive = 30 * time.Second
+
+//NewEmail makes email object for notifications and run sending daemon
+func NewEmail(params EmailParams) (*Email, error) {
+
+	res := Email{
+		server:    params.Server,
+		port:      params.Port,
+		from:      params.From,
+		username:  params.Username,
+		password:  params.Password,
+		keepAlive: params.KeepAlive,
+		sendChan:  make(chan *gomail.Message),
+		errChan:   make(chan error),
+	}
+
+	if res.keepAlive == 0 {
+		res.keepAlive = emailConnectionKeepAlive
+	}
+
+	log.Printf("[DEBUG] create new email notifier for server %s with user %s, keepalive=%s",
+		res.server, res.username, res.keepAlive)
+
+	// Test connection before starting a daemon.
+	tmpConn, err := gomail.NewDialer(res.server, res.port, res.username, res.password).Dial()
+	if err != nil {
+		return &res, errors.Wrapf(err, "error establishing test connecting to '%s':%d with username '%s'",
+			res.server, res.port, res.username)
+	}
+	if err = tmpConn.Close(); err != nil {
+		return &res, errors.Wrapf(err, "error closing test connection to %s:%d",
+			res.server, res.port)
+	}
+
+	// Activate server goroutine.
+	go res.activate()
+	// TODO: server goroutine never dies! This exiter should be run once first time we get context
+	// Close the channel to stop the mail daemon
+	// go func() { <-ctx.Done(); close(res.sendChan) }()
```

Comment:
I plan to defer server shutdown only after the first request: beforehand no one is waiting to send an email and there is nothing to shut down. Only after the first request, it makes sense to monitor context.
As it turns out, a program currently is not designed in a way which would allow bringing context to `NewEmail` (called from `makeNotify` called from `newServerApp`).